pkg/containerd/cmd/service/start.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/namespaces"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+)
+
+func start(args []string) {
+	invoked := filepath.Base(os.Args[0])
+	flags := flag.NewFlagSet("start", flag.ExitOnError)
+	flags.Usage = func() {
+		fmt.Printf("USAGE: %s start [service]\n\n", invoked)
+		fmt.Printf("Options:\n")
+		flags.PrintDefaults()
+	}
+
+	sock := flags.String("sock", "/run/containerd/containerd.sock", "Path to containerd socket")
+
+	dumpSpec := flags.String("dump-spec", "", "Dump container spec to file before start")
+
+	if err := flags.Parse(args); err != nil {
+		log.Fatal("Unable to parse args")
+	}
+	args = flags.Args()
+
+	if len(args) != 1 {
+		fmt.Println("Please specify the service")
+		flags.Usage()
+		os.Exit(1)
+	}
+
+	service := args[0]
+	rootfs := filepath.Join("/containers/services", service, "rootfs")
+	log.Infof("Starting service: %q", service)
+	log := log.WithFields(log.Fields{
+		"service": service,
+	})
+
+	client, err := containerd.New(*sock)
+	if err != nil {
+		log.WithError(err).Fatal("creating containerd client")
+	}
+
+	ctx := namespaces.WithNamespace(context.Background(), "default")
+
+	var spec *specs.Spec
+	specf, err := os.Open(filepath.Join("/containers/services", service, "config.json"))
+	if err != nil {
+		log.WithError(err).Fatal("failed to read service spec")
+	}
+	if err := json.NewDecoder(specf).Decode(&spec); err != nil {
+		log.WithError(err).Fatal("failed to parse service spec")
+	}
+
+	spec.Mounts = append([]specs.Mount{
+		{
+			Destination: "/",
+			Type:        "",
+			Source:      rootfs,
+			Options:     []string{"bind"},
+		},
+		{
+			Destination: "/",
+			Type:        "",
+			Source:      "",
+			Options:     []string{"remount", "rw"},
+		},
+	}, spec.Mounts...)
```

Comment:
I really don't like this, would rather do the remounts up front, not add them to the spec. These won't be needed if we use the image store, and they are not really part of the spec which should reflect what the user specified only, they are part of our implementation. Also they are only needed due to `runc` bugs.